### ISSUE TITLE
error on json encode for column with func as default

### ISSUE THREAD

**Issue by patvdleer** [thomaxxl/safrs#32]
I made a pretty simple mixin as seen below but the code seems to die on the dt.uctnow

```python
from datetime import datetime as dt


class CreatedMixin:
    created_at = db.Column(db.DateTime(timezone=True), default=dt.utcnow)
    updated_at = db.Column(db.DateTime(timezone=True), default=dt.utcnow, onupdate=dt.utcnow)

    @classmethod
    def newest(cls, amount=5):
        assert isinstance(cls, Model)
        return cls.query.order_by(cls.created_at.desc()).limit(amount)

    @classmethod
    def updated(cls, amount=5):
        assert isinstance(cls, Model)
        return cls.query.order_by(cls.updated_at.desc()).limit(amount)
```

![image](https://user-images.githubusercontent.com/1138136/54531882-aee1f500-4986-11e9-9270-b9c0998ac248.png)

stacktrace
```
127.0.0.1 - - [18/Mar/2019 14:00:46] "GET /swagger.json HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/_compat.py", line 34, in reraise
    raise value.with_traceback(tb)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/_compat.py", line 34, in reraise
    raise value.with_traceback(tb)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_debugtoolbar/__init__.py", line 125, in dispatch_request
    return view_func(**req.view_args)
  File "/usr/lib/python3.6/cProfile.py", line 109, in runcall
    return func(*args, **kw)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 484, in wrapper
    return self.make_response(data, code, headers=headers)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 513, in make_response
    resp = self.representations[mediatype](data, *args, **kwargs)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/lib/python3.6/site-packages/flask_restful/representations/json.py", line 21, in output_json
    dumped = dumps(data, **settings) + "\n"
  File "/usr/lib/python3.6/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.6/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/lib/python3.6/json/encoder.py", line 430, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.6/json/encoder.py", line 437, in _iterencode
    o = _default(o)
  File "/usr/lib/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'function' is not JSON serializable
```

**Comment by patvdleer** [comment]
It seems to be the default and not the onupdate, the code below does generate the `swagger.json`

```python
class CreatedMixin:
    created_at = db.Column(db.DateTime(timezone=True))
    updated_at = db.Column(db.DateTime(timezone=True), onupdate=dt.utcnow)
```

**Comment by thomaxxl** [comment]
Sorry for not answering earlier. I'll look into it.

I think this will work:
```python
created_at = db.Column(db.DateTime(timezone=True), default=dt.utcnow())
```

but that's not what you want obv. Will sort it out asap.

**Comment by patvdleer** [comment]
No worries, how was your vacation?

**Comment by thomaxxl** [comment]
lol, had a great time, that's why i couldn't reply indeed :)

**Comment by thomaxxl** [comment]
ok, implemented a fix, can you check it works for you?

**Comment by patvdleer** [comment]
Sadly no...

```python
from datetime import datetime as dt

# from flask_sqlalchemy import SQLAlchemy
from .extensions import db  # db = SQLAlchemy()

class CreatedMixin:
    created_at = db.Column(db.DateTime(timezone=True), default=dt.utcnow)
    updated_at = db.Column(db.DateTime(timezone=True), default=dt.utcnow, onupdate=dt.utcnow)
```

```
[2019-03-26 21:15:24,933] db:589 DEBUG: Failed to get python type for column publisher.created_at
[2019-03-26 21:15:24,934] _api:427 ERROR: <lambda>() missing 1 required positional argument: 'ctx'
Traceback (most recent call last):
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/src/safrs/safrs/_api.py", line 421, in api_decorator
    decorated_method = swagger_decorator(decorated_method)
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/src/safrs/safrs/swagger_doc.py", line 383, in swagger_doc_gen
    sample = cls._s_sample_dict()
  File "/home/patrick/PycharmProjects/ebookhub-backend/venv/src/safrs/safrs/db.py", line 592, in _s_sample_dict
    arg = column.default.arg()
TypeError: <lambda>() missing 1 required positional argument: 'ctx'
```

**Comment by thomaxxl** [comment]
Can you try now? The default arg is ignored now if it's callable.
I'll look at it again later, but callable defaults may cause more issues so I think it's better to just ignore them for now. 
The `_s_sample_dict` is used to generate a sample swagger payload. But when there's a column default, the default will be used so it's not really that important. and the swagger.json can be customized/extended anyway if needed.

**Comment by patvdleer** [comment]
The code does run now but throws a bunch of debug messages.

```
[2019-03-27 11:37:22,185] db:589 DEBUG: Failed to get python type for column publisher.created_at
[2019-03-27 11:37:22,186] db:589 DEBUG: Failed to get python type for column publisher.updated_at
[2019-03-27 11:37:22,251] db:589 DEBUG: Failed to get python type for column publisher.created_at
[2019-03-27 11:37:22,252] db:589 DEBUG: Failed to get python type for column publisher.updated_at
[2019-03-27 11:37:22,252] db:589 DEBUG: Failed to get python type for column publisher.created_at
[2019-03-27 11:37:22,253] db:589 DEBUG: Failed to get python type for column publisher.updated_at
```

**Comment by thomaxxl** [comment]
This message should be gone now.

**Comment by thomaxxl** [comment]
fixed in v2.1.0
